### ISSUE TITLE
[3.3.5]Spell: Enslave Demon

### ISSUE THREAD

**Issue by soniczhang11** [TrinityCore/TrinityCore#23582]
**Description:**

Bug with warlock's Enslave Demon

**Current behaviour:**

1. Warlocks can cast Enslave Demon, even they have not dismissed their pets.
2. Warlock cannot control enslaved demon to attack, but can let them follow, stop and use their spell.
3. Enslaved demons' spell have no gcd. For example, when you enslave a elite Felguard in Blasted Lands, it can spell Mortal Strike as fast as your button clicking.

**Expected behaviour:**

1. Warlocks can only ensalve Demon when their pets are already dismissed.
2. Enslaved demons should be able to attcak.
3. Enslaved demons' spell should have gcd and cd.

**Steps to reproduce the problem:**

1. .go c id 8717
2. enslave it
3. try above issues.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

4dbdc22b3b5770ff8252ea057bf7a7a92dcb31cb

**Operating system:** CentOS 7

**Comment by CraftedRO** [comment]
6a5fa255fe053796588684e3876f6047cf92a08f first issue is not true, on cata ptr you can Enslave Demon even if you have not dissmised your pet but the issue is that the current pet shoud be automatically dismissed doing so , instead on tc it will remain so we need a check on spell cast if owner have pet active to dismiss it https://youtu.be/C6eDzakx6B0

**Comment by Ovahlord** [comment]
yes, because charmed creatures and controlled creatures should not be mixed. that's a fundamental core issue which needs extensive reworks.
